src/sql/parts/connection/common/connectionProfileGroup.ts

```diff
@@ -153,6 +153,19 @@ export class ConnectionProfileGroup implements IConnectionProfileGroup {
 		return this.parent;
 	}
 
+	public isAncestorOf(node: ConnectionProfileGroup | ConnectionProfile): boolean {
+		let isAncestor = false;
+		let currentNode = node;
+		while (currentNode !== null && currentNode !== undefined) {
```

Comment:
nit: generally we write this as `while (currentNode) {`.  There are some minor difference between what you have and this, but usually they don't matter, and the simpler also syntax gets you what you want.